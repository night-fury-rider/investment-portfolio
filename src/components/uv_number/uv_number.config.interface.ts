export default interface UvNumberConfig {
  class ?: string;
  title ?: {
    class ?: string;
  };
  subTitle ?: {
    class ?: string;
  };
}
