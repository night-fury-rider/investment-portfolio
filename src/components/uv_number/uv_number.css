.uv-primary-text {
  font-size: 5rem;
}

.uv-number-container {
  height: 15rem;
}

@media screen and (max-width: 767px) {
  .uv-number-container {
    height: 15rem;
  }
}


.bounce-top {
	-webkit-animation: bounce-top 1s both;
	        animation: bounce-top 1s both;
}


/* ----------------------------------------------
 * Generated by Animista on 2020-11-17 11:20:53
 * Licensed under FreeBSD License.
 * See http://animista.net/license for more info. 
 * w: http://animista.net, t: @cssanimista
 * ---------------------------------------------- */

/**
 * ----------------------------------------
 * animation bounce-top
 * ----------------------------------------
 */
 @-webkit-keyframes bounce-top {
  0% {
    -webkit-transform: translateY(-45px);
            transform: translateY(-45px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
    opacity: 1;
  }
  24% {
    opacity: 1;
  }
  40% {
    -webkit-transform: translateY(-24px);
            transform: translateY(-24px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
  }
  65% {
    -webkit-transform: translateY(-12px);
            transform: translateY(-12px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
  }
  82% {
    -webkit-transform: translateY(-6px);
            transform: translateY(-6px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
  }
  93% {
    -webkit-transform: translateY(-4px);
            transform: translateY(-4px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
  }
  25%,
  55%,
  75%,
  87% {
    -webkit-transform: translateY(0px);
            transform: translateY(0px);
    -webkit-animation-timing-function: ease-out;
            animation-timing-function: ease-out;
  }
  100% {
    -webkit-transform: translateY(0px);
            transform: translateY(0px);
    -webkit-animation-timing-function: ease-out;
            animation-timing-function: ease-out;
    opacity: 1;
  }
}
@keyframes bounce-top {
  0% {
    -webkit-transform: translateY(-45px);
            transform: translateY(-45px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
    opacity: 1;
  }
  24% {
    opacity: 1;
  }
  40% {
    -webkit-transform: translateY(-24px);
            transform: translateY(-24px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
  }
  65% {
    -webkit-transform: translateY(-12px);
            transform: translateY(-12px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
  }
  82% {
    -webkit-transform: translateY(-6px);
            transform: translateY(-6px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
  }
  93% {
    -webkit-transform: translateY(-4px);
            transform: translateY(-4px);
    -webkit-animation-timing-function: ease-in;
            animation-timing-function: ease-in;
  }
  25%,
  55%,
  75%,
  87% {
    -webkit-transform: translateY(0px);
            transform: translateY(0px);
    -webkit-animation-timing-function: ease-out;
            animation-timing-function: ease-out;
  }
  100% {
    -webkit-transform: translateY(0px);
            transform: translateY(0px);
    -webkit-animation-timing-function: ease-out;
            animation-timing-function: ease-out;
    opacity: 1;
  }
}
