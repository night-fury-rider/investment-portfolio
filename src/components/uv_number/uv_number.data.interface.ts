export default interface UvNumberData {
  title : number;
  subtitle: string;
}
