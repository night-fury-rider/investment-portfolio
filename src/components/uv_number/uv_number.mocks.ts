export const numberTitle = 111;
export const numberSubtitle = 'Test Subtitle'
export const numberLabel = 'Test Label';
export const numberConfig = {
  class: 'test-component-class',
  title: {
    class: 'test-title-class'
  },
  subtitle: {
    class: 'test-subtitle-class'
  }
};
