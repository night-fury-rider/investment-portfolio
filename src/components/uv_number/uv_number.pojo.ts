import { UvNumberProps } from "../../shared/Types";

export default class UvNumberPojo {

  private _numberData: UvNumberProps;

  constructor(numberData: UvNumberProps) {
    this._numberData = {
      config: {
        class: (numberData.config && numberData.config.class) ? numberData.config.class : '',
        title: {
          class: (numberData.config && numberData.config.title && numberData.config.title.class) ?
                    numberData.config.title.class : '',
        },
        subtitle: {
          class: (numberData.config && numberData.config.subtitle && numberData.config.subtitle.class) ?
                    numberData.config.subtitle.class : '',
        }
      },
      data: {
        title: numberData.data.title,
        subtitle: numberData.data.subtitle ? numberData.data.subtitle : '',
        label: numberData.data.label ? numberData.data.label : ''
      }
    }
  }

  get numberData() {
    return this._numberData;
  }
}