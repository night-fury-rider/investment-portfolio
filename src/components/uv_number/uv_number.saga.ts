import { takeEvery } from "redux-saga/effects";
import UV_NUMBER from "./uv_number.constants";


function* loadUvNumber() {
  console.log('UV Number loaded');
  yield UV_NUMBER.LOAD;
}

export function* UvNumberSaga() {
  yield takeEvery(UV_NUMBER.LOAD, loadUvNumber);
}
