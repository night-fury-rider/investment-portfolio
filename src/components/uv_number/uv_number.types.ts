export type UVNumberProps = {
  config ?: {
    class ?: string;
    title ?: {
      class ?: string;
    };
    subtitle ?: {
      class ?: string;
    }
  };
  title: number;
  label: string;
  subtitle?: string;
};
