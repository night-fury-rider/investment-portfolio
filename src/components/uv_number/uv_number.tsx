import { memo } from 'react';
import { useSpring, animated } from 'react-spring';

import './uv_number.css';
import { UVNumberProps } from './uv_number.types';

const UVNumber = (props: UVNumberProps) => {

  // Note: number variable is from useSpring. Don't try to rename it.
  const { number } = useSpring({
    reset: true,
    from: { number: 0 },
    number: props.title,
    delay: 750
  });

  return (
    <div className = {'uv-number-container ' + ((props.config && props.config.class) ? props.config.class : '')}>
      <div className={'bounce-top uv-primary-text uv-text-center align-middle ' +
                              ((props.config && props.config.title && props.config.title.class ) ?
                                  props.config.title.class : '')}>
        <animated.div>{number.to(n => n.toFixed(2))}</animated.div>
      </div>

      <div className={'bounce-top uv-secondary-text uv-text-center ' +
              ((props.config && props.config.subtitle && props.config.subtitle.class) ? props.config.subtitle.class : '')}>
          {props.subtitle}
      </div>

      <div className="uv-component-label">{props.label}</div>

    </div>
  )
}

export default memo(UVNumber);
