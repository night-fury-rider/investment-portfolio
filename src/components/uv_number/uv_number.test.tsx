import { shallow } from 'enzyme';

import UVNumber from './uv_number';
import { numberTitle, numberLabel, numberConfig } from './uv_number.mocks';

let shallowComponent: any;

describe('UVNumber Component Suite', ()=> {

  beforeEach(()=> {
    shallowComponent = shallow(<UVNumber title={numberTitle} label={numberLabel}/>);
  });

  test('has correct snapshot', ()=> {
    expect(shallowComponent.find('.uv-number-container').length).toEqual(1);
  });

  test('renders title and label', ()=> {
    expect(shallowComponent.find('.uv-number-container').length).toEqual(1);
  });

  test('renders title and label with classes provided', ()=> {
    shallowComponent = shallow(<UVNumber title={numberTitle} label={numberLabel} config={numberConfig} />);
    expect(shallowComponent.find('.' + numberConfig.class).length).toEqual(1);
    expect(shallowComponent.find('.' + numberConfig.title.class).length).toEqual(1);
    expect(shallowComponent.find('.' + numberConfig.subtitle.class).length).toEqual(1);
  });
});
