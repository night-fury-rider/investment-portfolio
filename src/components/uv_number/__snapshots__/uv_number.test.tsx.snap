// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UVNumber Component Suite has correct snapshot 1`] = `
Object {
  "$$typeof": Symbol(react.memo),
  "compare": null,
  "type": [Function],
}
`;
